src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -654,20 +642,21 @@ private MsalAccessTokenCacheItem FilterByKeyId(MsalAccessTokenCacheItem item, Au
                                 "Filtering by home account id");
             }
 
+            // This will also filter for the case when familyId is null and exclude RTs with familyId in filtered list
             rtCacheItems = rtCacheItems.FilterWithLogging(item =>
-                            string.Equals(item.FamilyId ?? string.Empty,
-                            familyId ?? string.Empty, StringComparison.OrdinalIgnoreCase),
-                            requestParams.RequestContext.Logger,
-                            "Filtering by family id");
+                    string.Equals(item.FamilyId ?? string.Empty,
```

Comment:
Should this filter be in the `else` clause associated with `string.IsNullOrEmpty(familyId)` ? In the corner case that an FOCI app is used for OBO, all of the RTs from it will have familyId set to "1". OBO does not look for a family RT specifically, but when it looks for RT's for a specific client_id, it should not ignore an RT just because it also has familyId=1

Reply:
No, this should not be placed in the else block since this also filters RT for the foci refresh token. For example if an app left a family then it should not return a refresh token for that app. I added this to the else block earlier and there is a test that failed for this case: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/9d2c425e7b301580ee8355b6f8c6cf66f505384b/tests/Microsoft.Identity.Test.Unit/RequestsTests/FociTests.cs#L170